pkg/controller/migration/namespace_migration.go

```diff
@@ -469,35 +469,46 @@ func (m *CoreNamespaceMigration) getNodesToMigrate() []*v1.Node {
 // kubernetes rolling update.
 func (m *CoreNamespaceMigration) waitForCalicoPodsHealthy() error {
 	return wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
-		ksD, ksR, err := m.getNumPodsDesiredAndReady(kubeSystem, nodeDaemonSetName)
+		ksD, ksR, _, err := m.getNumPodsDesiredAndReady(kubeSystem, nodeDaemonSetName)
 		if err != nil {
 			return false, err
 		}
-		csD, csR, err := m.getNumPodsDesiredAndReady(common.CalicoNamespace, nodeDaemonSetName)
+		csD, csR, csMaxUnavailable, err := m.getNumPodsDesiredAndReady(common.CalicoNamespace, nodeDaemonSetName)
 		if err != nil {
 			return false, err
 		}
 
-		// TODO: When we support configuring adjusting the rolling update unavailable,
-		// we should use that configuration here.
-		if ksD == ksR && csD == csR {
+		// MaxUnavailable represents the number of pods that are allowed to be down during an update. We
+		// decrement the number of desired pods by that value.
```

Comment:
nit: this comment doesn't match what was actually done though logically it is the same.